### ISSUE TITLE
Docker-in-docker: lacks example?

### ISSUE THREAD

**Issue by qqgg231** [microsoft/vscode-dev-containers#178]
About "docker-in-docker" and similar:

Currently there is dockerfile in .devcontainer folder, which defines development container

However there is no example of dockerfile outside .devcontainer folder. Please add some dockerfile which, after deploying, writes to console / displays "hello world".

Such dockerfile should allow to create production container, not development container.

I assume here that such dockerfile should be separate, but maybe I am misunderstanding  things

**Comment by Chuxel** [comment]
@qqgg231 If you are looking for samples, the various "vscode-remote-try" repositories are a better fit.  For example https://github.com/Microsoft/vscode-remote-try-python

The contents of this used by VS Code for existing projects.  See [docs here](https://code.visualstudio.com/docs/remote/containers#_quick-start-open-an-existing-folder-in-a-container) for how this works.

In terms of reusing an existing Dockerfile, if you run the "Add Development Container Configuration File..." (or Reopen/Open Folder in Container) command, you will be presented with the option to reuse an existing Dockerfile if one is found.

The "Docker-in-Docker" [README](https://github.com/microsoft/vscode-dev-containers/tree/master/containers/docker-in-docker) includes information on how to adapt an existing Dockerfile if needed.

Does that address your questions?

**Comment by Chuxel** [comment]
Or are you specifically asking about a docker-in-docker example? (e.g. vscode-remote-try-docker-in-docker)

**Comment by qqgg231** [comment]
Yes, I think something like vscode-remote-try-docker-in-docker would be ok, so I and other people can see how to do it

**Comment by Chuxel** [comment]
@qqgg231 Got it - makes sense! Tweaked the title to reflect that.

**Comment by PavelSosin-320** [comment]
If so, busybox. It allows playing with it, mount volumes, connect to networks, use different drivers. attach, drop attach,  ...
